go-app-ussd_ccmdd_wc_address_update.js

```diff
@@ -253,6 +273,354 @@ go.app = (function() {
       });
     });
 
+    self.add("state_id_type", function(name) {
+      return new MenuState(name, {
+        question: $("What type of identification do you have?"),
+        error: $(
+          "Sorry we don't understand. Please enter the number next to your answer."
+        ),
+        choices: [
+          new Choice("state_sa_id_no", $("SA ID")),
+          new Choice("state_passport_country", $("Passport")),
+          new Choice("state_dob_year", $("None"))
+        ]
+      });
+    });
+
+    self.add("state_sa_id_no", function(name) {
+      return new FreeText(name, {
+        question: $(
+          "Please reply with your ID number as you find it in your Identity Document."
+        ),
+        check: function(content) {
+          var match = content.match(/^(\d{6})(\d{4})(0|1)8\d$/);
+          var today = new moment(self.im.config.testing_today).startOf("day"),
+            dob;
+          var validLuhn = function(content) {
+            return (
+              content
+                .split("")
+                .reverse()
+                .reduce(function(sum, digit, i) {
+                  return (
+                    sum +
+                    _.parseInt(
+                      i % 2 ? [0, 2, 4, 6, 8, 1, 3, 5, 7, 9][digit] : digit
+                    )
+                  );
+                }, 0) %
+                10 ==
+              0
+            );
+          };
+          if (
+            !match ||
+            !validLuhn(content) ||
+            !(dob = new moment(match[1], "YYMMDD")) ||
+            !dob.isValid() ||
+            !dob.isBetween(
+              today.clone().add(-130, "years"),
+              today.clone().add(-5, "years")
+            ) ||
+            _.parseInt(match[2]) >= 5000
```

Comment:
This will restrict the service to only female ID numbers. I'm not sure that we want to do that here.